compaction/compaction_strategy.cc

```diff
@@ -87,6 +87,34 @@ std::optional<sstring> compaction_strategy_impl::get_value(const std::map<sstrin
     return it->second;
 }
 
+void compaction_strategy_impl::validate_options(const std::map<sstring, sstring>& options, std::map<sstring, sstring>& unchecked_options) {
+    auto tmp_value = get_value(options, TOMBSTONE_THRESHOLD_OPTION);
+    auto tombstone_threshold = cql3::statements::property_definitions::to_double(TOMBSTONE_THRESHOLD_OPTION, tmp_value, DEFAULT_TOMBSTONE_THRESHOLD);
+    if (tombstone_threshold < 0.0 || tombstone_threshold > 1.0) {
+        throw exceptions::configuration_exception(fmt::format("{} value ({}) should be between between 0.0 and 1.0", TOMBSTONE_THRESHOLD_OPTION, tombstone_threshold));
+    }
+
+    db_clock::duration tombstone_compaction_interval = DEFAULT_TOMBSTONE_COMPACTION_INTERVAL();
+    tmp_value = get_value(options, TOMBSTONE_COMPACTION_INTERVAL_OPTION);
+
+    if (tmp_value) {
+        long interval;
+        try {
+            tombstone_compaction_interval = db_clock::duration(std::chrono::seconds(interval = std::stol(tmp_value.value())));
+        } catch (const std::exception& e) {
+            throw exceptions::syntax_exception(sstring("Invalid long value ") + tmp_value.value() + " for " + TOMBSTONE_COMPACTION_INTERVAL_OPTION);
+        }
+
+        if (interval <= 0) {
```

Comment:
why not check `tombstone_compaction_interval`?
can it become negative due to wrap around?